src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -365,10 +365,25 @@ public bool MatchesSymbol(ISymbol symbol)
 
             internal XElement CreateXElement()
                 => SymbolKind.HasValue ? new XElement(nameof(SymbolKind), SymbolKind) :
-                   TypeKind.HasValue ? new XElement(nameof(TypeKind), TypeKind) :
+                   TypeKind.HasValue ? new XElement(nameof(TypeKind), GetTypeKindString(TypeKind.Value)) :
                    MethodKind.HasValue ? new XElement(nameof(MethodKind), MethodKind) :
                    throw ExceptionUtilities.Unreachable;
 
+            private static string GetTypeKindString(TypeKind typeKind)
+            {
+                // We have two members in TypeKind that point to the same value, Struct and Structure. Because of this,
+                // Enum.ToString(), which under the covers uses a binary search, isn't stable which one it will pick and it can
+                // change if other TypeKinds are added. This ensures we keep using the same string consistently.
+                if (typeKind == CodeAnalysis.TypeKind.Structure)
+                {
+                    return nameof(CodeAnalysis.TypeKind.Struct);
```

Comment:
seems fragile that we only handle this case.  can we add some sort of static check and find the dupes and assert if a new dupe happens in the futuer that we aren't handling?

Reply:
Good idea, will add the test.

Reply:
(I also hope we never do this again.)